search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -95,20 +94,27 @@ class TemporalEntityHandler(
                 .body(BadRequestDataResponse(e.message))
         }
 
-        val temporalEntityAttribute = temporalEntityAttributeService.getForEntity(
+        val temporalEntityAttributes = temporalEntityAttributeService.getForEntity(
             entityId.toUri(),
             temporalQuery.attrs,
             contextLink
-        ).awaitFirstOrNull() ?: throw ResourceNotFoundException(entityNotFoundMessage(entityId))
+        ).collectList().awaitFirst()
 
-        val results = attributeInstanceService.search(temporalQuery, temporalEntityAttribute).awaitFirst()
+        if (temporalEntityAttributes.isEmpty()) throw ResourceNotFoundException(entityNotFoundMessage(entityId))
```

Comment:
Can you try this ?
`.collectList().awaitFirst().ifEmpty {throw ResourceNotFoundException(entityNotFoundMessage(entityId))}`

Reply:
Yes, didn't know that one, it's handy